src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/message/CreateUpdateService.kt

```diff
@@ -26,40 +27,40 @@ import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
 class CreateUpdateService(
   private val telemetryService: TelemetryService,
   private val personService: PersonService,
-  private val personKeyService: PersonKeyService,
   private val readWriteLockService: ReadWriteLockService,
-  @Value("\${retry.delay}") private val retryDelay: Long,
+  private val queueService: QueueService,
   private val eventLoggingService: EventLoggingService,
-
+  @Value("\${retry.delay}") private val retryDelay: Long,
 ) {
 
-  fun processMessage(person: Person, event: String? = null, linkRecord: Boolean = true, callback: () -> PersonEntity?): PersonEntity = runBlocking {
+  fun processMessage(person: Person, event: String? = null, callback: () -> PersonEntity?): PersonEntity = runBlocking {
     runWithRetry(MAX_ATTEMPTS, retryDelay, ENTITY_RETRY_EXCEPTIONS) {
       readWriteLockService.withWriteLock(person.sourceSystemType) {
-        return@withWriteLock processPerson(person, event, linkRecord, callback)
+        return@withWriteLock processPerson(person, event, callback)
       }
     }
   }
 
-  private fun processPerson(person: Person, event: String?, linkRecord: Boolean, callback: () -> PersonEntity?): PersonEntity {
-    val existingPersonEntity: PersonEntity? = callback()
-    val personEntity: PersonEntity = when {
-      (existingPersonEntity == null) -> handlePersonCreation(person, event, linkRecord)
-      else -> handlePersonUpdate(person, existingPersonEntity, event)
-    }
-    return personEntity
+  @Transactional
```

Comment:
https://docs.spring.io/spring-framework/reference/data-access/transaction/declarative/annotations.html
This doesn't do anything - @Transactional is only effective on methods invoked from other classes